### DISCUSSION TITLE
Add `Response.ajson`

### DISCUSSION THREAD

**Discussion by ToxicKidz** [encode/httpx#1639] [Ideas] [answered]
Since there's already `Resonse.json()` and `Response.aread()`, why not have `Response.ajson`?

## Current Solution
```py
data = json.loads(await response.aread())

# OR

await response.aread()
data = response.json()
```
## Implementation
The implementation would be fairly easy:
```py
class Response:
    ...
    async def ajson(self, **kwargs: Any) -> Any:
        await self.aread()
        return self.json(**kwargs)
```

**Comment by lovelydinosaur** [comment, marked as answer]
So, this wouldn’t make sense as a counterpart because `.json()` *doesn’t* read the response body.

Let’s explain that more clearly. There’s no need for an `ajson()`, because this already works just fine...

```python
r = await client.get(...)
r.json()  # There’s no I/O occurring here.
```

The only case where you need to explicitly perform I/O is in the streaming case.

```python
async with client.stream(...) as r:
    await r.aread()
    r.json()  # Can only call .json once we’ve loaded the response body.
```

Which anyway matches the style you’d need to use in the sync case...

```python
with client.stream(...) as r:
    r.read()
    r.json()  # Can only call .json once we’ve loaded the response body.
```
